### PR TITLE
[5455] Revert qts awards and show award reversal in timeline

### PR DESCRIPTION
**Pull request by markhallen**
### Context

In this pull request, we address an issue where 127 trainees were mistakenly marked as awarded by their providers. The proposed changes aim to revert this incorrect state and ensure that the reversal is properly reflected in the trainee timeline.

### Changes proposed in this pull request

The following changes are included in this pull request:

1. Data Migration:
   - A data migration is added to remove the award state for a specific list of trainee slugs. This migration will correct the mistakenly awarded trainees by removing the award status from their records.

2. Logic for Handling and Displaying Reversed Awards:
   - The necessary logic is implemented to handle and display reversed awards correctly. This ensures that the trainee timeline accurately reflects the reversal of the mistakenly awarded status.

### Guidance to review

To review the changes in this pull request, please follow these steps:

1. Run the migration against production-like data:
   - Execute the provided data migration script against a production-like dataset to verify its correctness and effectiveness in removing the award state for the identified trainees.
   - `bin/rails data:migrate`

2. Check the timeline of an updated trainee:
   - Select a trainee from the affected list and review their timeline. Ensure that the reversal of the awarded status is accurately displayed and that the timeline reflects the corrected state.
   - Test on 127.0.0.1 - [nPsk5npdB2wiJNrYUkdAHuSG](http://127.0.0.1:5000/trainees/nPsk5npdB2wiJNrYUkdAHuSG/timeline)

By carefully reviewing and validating these changes, we can ensure that the issue of mistakenly awarded trainees is resolved, and the trainee timeline accurately represents the corrected state.

#### Timeline

image

With subsequent deferral

image

### Important business

- [x] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Comment by markhallen** [comment]
[[S] Update trainee timeline to handle undo of awards](https://trello.com/c/XSzpqW3W/5505-s-update-trainee-timeline-to-handle-undo-of-awards)

**Comment by markhallen** [comment]
[Revert QTS Awards](https://trello.com/c/bpOpoG47/5455-revert-qts-awards)

File: app/services/trainees/create_timeline_events.rb [hunk 1]
```diff
@@ -132,22 +135,28 @@ def call
...
+      change.is_a?(Array) &&
+        change.size == 2 &&
+        change[0] == Trainee.states[state] &&
+        change[1].is_a?(Integer)
     end
 
-    def undo_withdraw_message
-      [["Comment:", undo_withdraw_comment]] if comment.present?
+    def undo_message
+      [["Comment:", undo_comment]] if comment.present?
```
Comment (tavy88) [inline app/services/trainees/create_timeline_events.rb:156]: Should this not be flatten?

**Comment by markhallen** [comment]
[Move trainees accidentally awarded QTS back to TRN received.](https://trello.com/c/8fYAvojT/5507-move-trainees-accidentally-awarded-qts-back-to-trn-received)

**Pull request merged**